diff --git a/node_modules/jsbeeb/src/utils.js b/node_modules/jsbeeb/src/utils.js
index c594053..84d8f26 100644
--- a/node_modules/jsbeeb/src/utils.js
+++ b/node_modules/jsbeeb/src/utils.js
@@ -841,11 +841,16 @@ function loadDataHttp(url) {
 }
 
 let _nodeBasePath = null;
+let urlMap = {};
 
 export function setNodeBasePath(basePath) {
     _nodeBasePath = basePath;
 }
 
+export function setUrlMap(map) {
+    urlMap = map;
+}
+
 async function loadDataNode(url) {
     if (typeof readbuffer !== "undefined") {
         // d8 shell
@@ -871,6 +876,7 @@ async function loadDataNode(url) {
 }
 
 export function loadData(url) {
+    url = urlMap[url] ?? url;
     if (runningInNode) {
         return loadDataNode(url);
     } else {
